internal/criapi/criapi.go

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2021. Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+
+package criapi
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+	"os"
+	"time"
+)
+
+// Client represents the client grpc connection to the ContainerRuntimerInterface
+type Client struct {
+	CRIConn              *grpc.ClientConn              // A grpc client connection to CRI
+	RuntimeServiceClient v1alpha2.RuntimeServiceClient // A RuntimeService climent
+}
+
+// CRIClient is an intstance of the Client for the CRI connection
+var CRIClient Client
+
+// CRIClientDialRetry is the amount of time to wait before retrying
+var CRIClientDialRetry = 30 * time.Second
+
+// CRIMaxConnectionRetry is the maximum number of connection retries.
+var CRIMaxConnectionRetry = 3
+
+// NewCRIClient returns a new client connection to the ContainerRuntimeInterface or an error
+func NewCRIClient(criSock string, clientOpts ...grpc.DialOption) (*Client, error) {
+	var err error
+	ctx, cancel := context.WithTimeout(context.Background(), 90*time.Second)
```

Comment:
Suggestion: make the `90*time.Second` a const or var?

Reply:
fixed in next push.